addons/auto-follow-topics/userscript.js

```diff
@@ -0,0 +1,13 @@
+export default async function ({ addon, console, msg }) {
+  document.querySelector("[name=AddPostForm]").addEventListener("click", (event) => {
+    localStorage.setItem(location.href.split("/")[5], true);
+  });
+
+  if (localStorage.getItem(location.href.split("/")[5])) {
+    console.log("Following topic");
+    const followBtn = document.querySelectorAll(".follow-button")[1];
+    followBtn.focus();
+    followBtn.click();
+    localStorage.clear();
```

Comment:
Would this clear everything in local storage and possibly affect Scratch, or...?

Reply:
Yeah, please avoid clearing all local storage. Use `localStorage.removeItem("key")` instead.

Reply:
The root cause of the problem is that the local storage key is being used to store information, not just as an identifier. That's a pattern to avoid.
@Chiroyce I'd suggest having a single local storage key, with a constant name, that stores an array of IDs in JSON format.

Reply:
My bad, did not consider that. I *somehow* thought that every addon had unique storage (Note to self: Always read docs)

> The root cause of the problem is that the local storage key is being used to store information, not just as an identifier. That's a pattern to avoid.

Got that, will make a fixed key like `auto-follow-topics`


> @Chiroyce I'd suggest having a single local storage key, with a constant name, that stores an array of IDs in JSON format.

A small problem with that I can think of, if a user unfollows a topic and revisits that page, it would follow it again. So I think its best not to make it persistent, but instead just check if a user has posted in a topic just before the page reloaded